service/pagerduty/pager_duty.go

```diff
@@ -0,0 +1,70 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/PagerDuty/go-pagerduty"
+	"github.com/nikoksr/notify"
+)
+
+type Client interface {
+	CreateIncidentWithContext(ctx context.Context, from string, options *pagerduty.CreateIncidentOptions) (*pagerduty.Incident, error)
+}
+
+// Compile-time check to verify that the PagerDuty type implements the notifier.Notifier interface.
+var _ notify.Notifier = &PagerDuty{}
+
+type PagerDuty struct {
+	*Config
+	Client Client
+}
+
+func New(token string, clientOptions ...pagerduty.ClientOptions) (*PagerDuty, error) {
+	if token == "" {
+		return nil, fmt.Errorf("access token is required")
+	}
+
+	pagerDuty := &PagerDuty{
+		Config: NewConfig(),
+		Client: pagerduty.NewClient(token, clientOptions...),
+	}
+
+	return pagerDuty, nil
+}
+
+func (s *PagerDuty) Send(ctx context.Context, subject, message string) error {
+	if err := s.Config.OK(); err != nil {
+		return fmt.Errorf("invalid configuration: %w", err)
+	}
+
+	incident := s.IncidentOptions(subject, message)
+
+	for _, receiver := range s.Config.Receivers {
+		// set the service ID to the receiver
+		incident.Service.ID = receiver
+
+		_, err := s.Client.CreateIncidentWithContext(ctx, s.Config.FromAddress, incident)
+		if err != nil {
+			return fmt.Errorf("failed to create pager duty incident: %w", err)
```

Comment:
```suggestion
			return fmt.Errorf("create pager duty incident: %w", err)
```

I'd prefer this to avoid redundant `failed to` prefixes in the final error. Since the error will most likely be nested, prefixing it here could lead to an error message like:

```bash
failed to send important message to user: send notification: failed to create pager duty incident: <actual error>
```

I'd prefer reading this:

```bash
failed to send important message to user: send notification: create pager duty incident: <actual error>
```